[DOC-7251] Private clusters and CMEK are GA (#16553)
cockroachdb/docs: cockroachcloud/private-clusters.md

By default, {{ site.data.products.db }} has safeguards in place to protect cluster's data from the public internet. Ingress traffic to a cluster is routed through a load balancer, and it is possible to restrict inbound connections using a combination of [IP allowlisting](/docs/cockroachcloud/network-authorization.html#ip-allowlisting), and either of [AWS PrivateLink](/docs/cockroachcloud/network-authorization.html#aws-privatelink) or [GCP VPC peering](/docs/cockroachcloud/network-authorization.html#vpc-peering) depending on your cloud provider. However, data egress operations such as [exports](/docs/stable/export.html), [backups](/docs/stable/backup.html), and [Change Data Capture (CDC)](/docs/stable/change-data-capture-overview.html) use public subnets.

On the other hand, a private cluster's nodes have no public IP addresses, and egress traffic moves over private subnets and through a highly-available NAT gateway that is unique to the cluster.

> One private network exists per cluster region, and each node is connected to the private network for its region. A highly-available NAT gateway connects the private networks and provides a static egress public IP address. On AWS, the NAT gateways are created in three separate availability zones to mitigate against the risk of an availability zone outage.

Reviewer: ```suggestion
One private network exists per cluster region, and each node is connected to the private network for its region. A NAT gateway is connected to each private network and provides a static egress public IP address.
```

> Egress traffic from the cluster nodes to S3 or Google Cloud Storage flows across the private subnet and through the private network. All egress traffic from the cluster nodes to non-cloud external resources flows across the private subnet and through the NAT gateway.

Reviewer: ```suggestion
Egress traffic from the cluster nodes to S3 or Google Cloud Storage flows across the private subnet and through the cloud provider's private network. Egress traffic from the cluster nodes to all other external resources flows across the private subnet and through the NAT gateway.
```

This page shows how to create a private cluster.

## Create a private cluster

> When you [create](https://cockroachlabs.com/docs/api/cloud/v1/clusters) a private cluster, you set its `private_network_visibility` field to `true`. This in turn automatically sets `network_visibility` to `NETWORK_VISIBILITY_PRIVATE`. To create a private cluster, you must use [{{ site.data.products.db }} API](cloud-api.html) or [CockroachDB's Terraform provider](https://registry.terraform.io/providers/cockroachdb/cockroach/latest/docs/resources/cluster).

Reviewer: ```suggestion
To create a private cluster, you must use [{{ site.data.products.db }} API](cloud-api.html) or [CockroachDB's Terraform provider](https://registry.terraform.io/providers/cockroachdb/cockroach/latest/docs/resources/cluster). When you [create](https://cockroachlabs.com/docs/api/cloud/v1/clusters) a private cluster using Terraform Provider, you set its `private_network_visibility` field to `true`. This in turn automatically sets Cloud API's `network_visibility` to `NETWORK_VISIBILITY_PRIVATE`.
```

## Create a private cluster

- By default, newly-created {{ site.data.products.dedicated }} clusters deployed on GCP will be private clusters.
- By default, newly-created {{ site.data.products.dedicated }} clusters deployed on AWS will **not** be private clusters.

>   To create a private cluster on AWS, you must use [{{ site.data.products.db }} API](cloud-api.html), When you [create the cluster](https://cockroachlabs.com/docs/api/cloud/v1/clusters), set the set the `network_visibility` field to `NETWORK_VISIBILITY_PRIVATE`. If you use [CockroachDB's Terraform provider](https://registry.terraform.io/providers/cockroachdb/cockroach/latest/docs/resources/cluster), set `private_network_visibility` to `true`, which in turn sets `network_visibility` to `NETWORK_VISIBILITY_PRIVATE`.

Reviewer: ```suggestion
  To create a private cluster on AWS, you must use [{{ site.data.products.db }} API](cloud-api.html) or Terraform Provider(https://registry.terraform.io/providers/cockroachdb/cockroach/latest). When you [create the cluster](https://cockroachlabs.com/docs/api/cloud/v1/clusters), set the `network_visibility` field to `NETWORK_VISIBILITY_PRIVATE`. If you use [CockroachDB's Terraform provider](https://registry.terraform.io/providers/cockroachdb/cockroach/latest/docs/resources/cluster), set `private_network_visibility` to `true`, which in turn sets `network_visibility` to `NETWORK_VISIBILITY_PRIVATE`.
```

> - An existing cluster can't be migrated in-place to a private cluster.

Reviewer 2: The structure of this section is a little bit difficult to read. It seems like these are mostly considerations when creating a private cluster, and the actual instructions are buried between bullet points. Maybe the first half could be two short paragraphs for GCP and AWS, and the second half after "When you create a private cluster:.." could be a bulleted list, not numbered steps. Just a suggestion

Author: Updated this section a bit. Thanks.
